src/SwapExtractionTool/VoteExtractionService.cs

```diff
@@ -131,11 +136,33 @@ private async Task ProcessBlockForCollateralVoteTransactionsAsync(BlockTransacti
                     }
 
                     AddressBalancesResult balance = await $"http://localhost:{this.StratisNetworkApiPort}/api"
-                            .AppendPathSegment("blockstore/getaddressesbalances")
-                            .SetQueryParams(new { addresses = potentialStratAddress, minConfirmations = 0 })
-                            .GetJsonAsync<AddressBalancesResult>();
+                        .AppendPathSegment($"blockstore/{BlockStoreRouteEndPoint.GetAddressesBalances}")
+                        .SetQueryParams(new { addresses = potentialStratAddress, minConfirmations = 0 })
+                        .GetJsonAsync<AddressBalancesResult>();
+
+                    Money determinedBalance = balance.Balances[0].Balance;
+
+                    // Check if the last transaction that spends from the given address was a burn transaction.
+                    // If so, the amount of the burn takes precedence for the purposes of the vote weight.
+                    LastBalanceDecreaseTransactionModel lastBalanceDecreaseTransaction = await $"http://localhost:{this.StratisNetworkApiPort}/api"
+                        .AppendPathSegment($"blockstore/{BlockStoreRouteEndPoint.GetLastBalanceDecreaseTransaction}")
+                        .SetQueryParams(new { addresses = potentialStratAddress, minConfirmations = 0 })
+                        .GetJsonAsync<LastBalanceDecreaseTransactionModel>();
+
+                    if (lastBalanceDecreaseTransaction.BlockHeight > blockHeight)
+                    {
+                        foreach (Vout txOut in lastBalanceDecreaseTransaction.Transaction.VOut)
+                        {
+                            if (txOut.ScriptPubKey.Type == "nulldata" && Money.Coins(txOut.Value) > determinedBalance)
```

Comment:
I'm wondering whether we should also check the burn transaction itself to see if it does contain a STRAX address? Maybe a bit overkill? 🤔